sources/packages/web/src/services/InstitutionUserService.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  InstitutionUserRoles,
+  InstitutionUserTypes,
+  LocationAuthorization,
+  LocationUserAccess,
+  InstitutionUserViewModel,
+  InstitutionUserSummary,
+  PaginationOptions,
+} from "@/types";
+import ApiClient from "@/services/http/ApiClient";
+import { AuthService } from "@/services/AuthService";
+import {
+  InstitutionUserAPIOutDTO,
+  CreateInstitutionUserAPIInDTO,
+  UpdateInstitutionUserAPIInDTO,
+  UserPermissionAPIInDTO,
+  InstitutionUserStatusAPIOutDTO,
+  InstitutionUserDetailAPIOutDTO,
+} from "@/services/http/dto";
+
+export class InstitutionUserService {
+  // Share Instance
+  private static instance: InstitutionUserService;
+
+  static get shared(): InstitutionUserService {
+    return this.instance || (this.instance = new this());
+  }
+
+  /**
+   * Get details of user who is logged in.
+   * @returns user details.
+   */
+  async getMyInstitutionDetails(): Promise<InstitutionUserDetailAPIOutDTO> {
+    return ApiClient.InstitutionUserApi.getMyInstitutionDetails();
+  }
+
+  /**
+   * Convert the institution user preparing it to be displayed as a summary view.
+   * @param institutionUser institution user to be converted.
+   * @returns institution user prepared to be displayed on the UI.
+   */
+  private convertToInstitutionUserViewModel(
+    institutionUser: InstitutionUserAPIOutDTO,
+  ): InstitutionUserViewModel {
+    const roles = institutionUser.authorizations
+      .filter((user) => !!user.authType.role)
+      .map((user) => user.authType.role);
+    const userTypes = institutionUser.authorizations.map(
+      (user) => user.authType.type,
+    );
+    const isAdmin = userTypes.includes(InstitutionUserTypes.admin);
+    const userType = isAdmin
+      ? InstitutionUserTypes.admin
+      : InstitutionUserTypes.user;
+    const locations: string[] = [];
+    if (isAdmin) {
+      locations.push("All");
+    } else {
+      const locationNames = institutionUser.authorizations
+        .filter((authorization) => !!authorization.location)
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
do we need to disable this? can't we use `authorization.location?.name`

Reply:
Without the "!" it is not able to recognize that the '.filter. is already ensuring that all the locations will have a name.
![image](https://user-images.githubusercontent.com/61259237/179324561-2210a4d5-9302-48cb-8b66-ab3a8a67305c.png)

Reply:
mm, got from the `typescript-eslint`. tried this?
![image](https://user-images.githubusercontent.com/77353155/179325473-33f759b0-05ce-4118-a125-9ed9f1ae07eb.png)

Reply:
Changed as per recommendation. 
![image](https://user-images.githubusercontent.com/61259237/179326092-58034136-6581-4045-9d56-4855375ae698.png)